src/gurkha.js

```diff
@@ -0,0 +1,240 @@
+
+    return resultArray
+  }
+
+  _parseObject ($currentElement, sch, sanitizer) {
+    const $ = this._$
+    const rule = sch.$rule
+    const resultArray = []
+    const anonymous = () => { return false }
+    // options
+    const topLevel = sch.$topLevel
+    const post = sch.$post
+    const ignore = sch.$ignore || anonymous
+    const constant = sch.$constant
+
+    if (constant) {
+      return constant
+    } else {
+      if (rule) {
+        if (typeof rule !== 'string') {
+          throw new Error('Illegal type: Rules must be in String format')
+        }
+
+        if (!$currentElement || topLevel) {
+          $currentElement = $(rule)
+        } else {
+          $currentElement = $($currentElement).find(rule)
+        }
+
+        $currentElement.each((index, el) => {
+          const $el = $(el)
+
+          if (typeof ignore !== 'function') {
+            throw new Error('Illegal type: Filters must be in function format')
+          }
+
+          if (!ignore($el, this._extvars)) {
+            resultArray.push(this._build($el, sch, sanitizer))
+          }
+        })
+      } else {
+        if (!$currentElement || topLevel) {
+          $currentElement = $(this._html)
+        }
+
+        resultArray.push(this._build($currentElement, sch, sanitizer))
+      }
+
+      // post-processing
+      if (post) {
+        if (typeof post !== 'function') {
+          throw new Error('Illegal type: Post-processing functions must be in function format')
+        } else {
+          return resultArray.map((result) => {
+            return post(this._flatten2(result), this._extvars)
+          })
+        }
+      } else {
+        return resultArray
+      }
+    }
+  }
+
+  _parseString ($currentElement, sch, sanitizer) {
+    const $ = this._$
+    const resultArray = []
+    let $subElement
+
+    if (!$currentElement) {
+      $subElement = $(sch)
+    } else {
+      $subElement = $currentElement.find(sch)
+    }
+
+    $subElement.each((index, el) => {
+      const $el = $(el)
+
+      if (!sanitizer) {
+        resultArray.push($el.text())
+      } else {
+        resultArray.push(sanitizer($el, this._extvars))
+      }
+    })
+
+    return resultArray
+  }
+
+  _build ($el, sch, sanitizer) {
+    let value
+    const result = {}
+    let keyCount = 0
+    sanitizer = sch.$sanitizer || sanitizer
+
+    if (sanitizer) {
+      if (typeof sanitizer !== 'function') {
+        throw new Error('Illegal type: Sanitizers must be in function format')
+      }
+    }
+
+    for (let key in sch) {
+      if (this._reserved[key] || !key) {
+        continue
+      } else {
+        keyCount += 1
+        value = sch[key]
+        result[key] = this._parse($el, value, sch.$sanitizer)
+      }
+    }
+
+    if (keyCount === 0) {
+      if (sanitizer) {
+        return sanitizer($el, this._extvars)
+      } else {
+        return $el.text()
+      }
+    } else {
+      return result
+    }
+  }
+
+  _flatten (val) {
+    const array = []
```

Comment:
Technically it is not constant, the same for the object below (line: 208 and others more)
I know it works, but does not look good constant variable that can be modified :P